relay-server/src/actors/processor.rs

```diff
@@ -2156,6 +2174,9 @@ impl EnvelopeProcessorService {
             });
         }
 
+        // We need the event parsed in order to set the profile context on it
+        self.process_profiles(state);
```

Comment:
Most of this function is constrained to processing mode. Would it make sense to:
 - Have a small function checking for the profiling feature flag above
 - Make the rest `#[cfg(feature = "processing")]` and call it after `enforce_quotas` inside `if_processing`
 
 ?

Reply:
Sounds good to me.